Pirat.DatabaseTests/ResourceChangeTest.cs

```diff
@@ -280,5 +280,175 @@ public async void Test_ChangePersonalInformation_NotPossible()
+            Consumable changedConsumable = (await _resourceDemandService.queryLink(_token)).consumables[0];
+            Assert.Equal(newAmount, changedConsumable.amount);
+            
+            // Consumable, with a reason
+            newAmount += 99;
+            await _resourceUpdateService.ChangeConsumableAmount(_token, consumable.id, newAmount, 
+                "This is an unnecessary reason.");
+            changedConsumable = (await _resourceDemandService.queryLink(_token)).consumables[0];
+            Assert.Equal(newAmount, changedConsumable.amount);
+        }
+
+        /// <summary>
+        /// Tests if the amount of devices and consumables can be decreased when a reason if provided.
+        /// </summary>
+        [Fact(Skip = "TODO")]
+        public async void Test_DecreaseDeviceOrConsumableAmount_Possible()
+        {
+            Device device = _offer.devices[0];
+            Consumable consumable = _offer.consumables[0];
+            
+            // Device
+            var newAmount = device.amount - 2;
+            await _resourceUpdateService.ChangeDeviceAmount(_token, device.id, newAmount,
+                "Given away with the help of PIRAT");
+            Device changedDevice = (await _resourceDemandService.queryLink(_token)).devices[0];
+            Assert.Equal(newAmount, changedDevice.amount);
+            
+            // Consumable
+            newAmount = consumable.amount - 5;
+            await _resourceUpdateService.ChangeConsumableAmount(_token, consumable.id, newAmount, "Eaten by a shark");
+            Consumable changedConsumable = (await _resourceDemandService.queryLink(_token)).consumables[0];
+            Assert.Equal(newAmount, changedConsumable.amount);
+        }
+
+        /// <summary>
+        /// Tests that the amount cannot be decreased if no reason is given.
+        /// </summary>
+        [Fact(Skip = "TODO")]
+        public async void Test_DecreaseDeviceOrConsumableAmount_MissingReason_Error()
+        {
+            Device device = _offer.devices[0];
+            Consumable consumable = _offer.consumables[0];
+            
+            // Device, missing reason
+            var newAmount = device.amount - 2;
+            await Assert.ThrowsAnyAsync<Exception>(() => _resourceUpdateService
+                .ChangeDeviceAmount(_token, device.id, newAmount, ""));
+            Assert.Equal(device.amount,
+                (await _resourceDemandService.queryLink(_token)).devices[0].amount);
+
+            // Consumable, missing reason
+            newAmount = consumable.amount - 5;
+            await Assert.ThrowsAnyAsync<Exception>(() => _resourceUpdateService
+                .ChangeConsumableAmount(_token, consumable.id, newAmount));
+            Assert.Equal(consumable.amount,
+                (await _resourceDemandService.queryLink(_token)).consumables[0].amount);
+        }
+
+        /// <summary>
+        /// Tests that the amount cannot be decreased to 0.
```

Comment:
Es könnte doch ein Offer geben, in dem ein Consumable gematched wurde und damit auf 0 geht, aber der Rest des Offers weiterhin bestehen soll.

Reply:
Dafür soll eine eigene delete Methode verwendet werden.